From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Kasiewicz, Marek" <marek.kasiewicz@intel.com>
Date: Wed, 9 Apr 2026 09:36:00 +0000
Subject: [PATCH] ice: e830: use direct MMIO for PHY timer command

On E830, ice_ptp_port_cmd_e830() sends the PHY timer command
(E830_ETH_GLTSYN_CMD at 0x00088814) via sideband queue messaging
(ice_write_phy_reg_e810_lp). This path was designed for E810 which
has an external PHY requiring SBQ access, but E830 has an integrated
PHY whose registers are directly accessible via BAR0 MMIO.

The SBQ path fails with ICE_ERR_AQ_ERROR (-100), which causes
ice_ptp_adj_clock() to fail, breaking PTP clock synchronization.

Other E830 PTP functions already use direct MMIO:
  - ice_ptp_write_direct_phc_time_e830() uses wr32(GLTSYN_TIME_*)
  - ice_ptp_write_direct_incval_e830() uses wr32(GLTSYN_INCVAL_*)
  - ice_ptp_src_cmd() uses wr32(GLTSYN_CMD) at 0x00088810

E830_ETH_GLTSYN_CMD (0x00088814) is in the same register space,
4 bytes after GLTSYN_CMD (0x00088810), and is directly accessible.

Replace the SBQ call with a direct wr32() to match the pattern
used by all other E830 PTP register accesses.

Signed-off-by: Kasiewicz, Marek <marek.kasiewicz@intel.com>
---
 drivers/net/intel/ice/base/ice_ptp_hw.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/intel/ice/base/ice_ptp_hw.c b/drivers/net/intel/ice/base/ice_ptp_hw.c
index 1234567890..abcdef1234 100644
--- a/drivers/net/intel/ice/base/ice_ptp_hw.c
+++ b/drivers/net/intel/ice/base/ice_ptp_hw.c
@@ -5627,8 +5627,9 @@ ice_ptp_port_cmd_e830(struct ice_hw *hw, enum ice_ptp_tmr_cmd cmd,
 {
 	u32 val = ice_ptp_tmr_cmd_to_port_reg(hw, cmd);
 
-	return ice_write_phy_reg_e810_lp(hw, E830_ETH_GLTSYN_CMD, val,
-					   lock_sbq);
+	wr32(hw, E830_ETH_GLTSYN_CMD, val);
+
+	return 0;
 }
 
 /**
-- 
2.34.1
